compiler/commands.nim

```diff
@@ -823,6 +821,7 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
   of "skipparentcfg":
     processOnOffSwitchG(conf, {optSkipParentConfigFiles}, arg, pass, info)
   of "genscript", "gendeps":
+    if switch.normalize == "gendeps": deprecatedAlias(switch, "genscript")
```

Comment:
because --fullhelp only mentions genscript, nim repo only uses genscript, and this is a case of pointless aliasing

Reply:
I think "gendeps" used to do something else but ok.

Reply:
gendeps was introduced in da90657317e8a57bae80ebd2d637a972d3b438ab ;  the intent was probably just to call `writeDepsFile` but instead it was an alias of genscript (with misleading help msg).
We could re-introduce this under a different name and more flexible, eg:
```
nim c --dump main
```
writes to `main.json` a json file with detailed information about the compilation process (extending hintsuccessx), including dependencies.
`--dump` would differ from `dump` command in that it'd be post-compilation